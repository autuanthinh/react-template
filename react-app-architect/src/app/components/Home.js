import React, { PureComponent } from 'react';

class Home extends PureComponent{

    render(){
        return(
            <div className="home">
                <p>Home</p>
                <div className="cssTest">test css</div>
            </div>
        );
    }
}

export default Home;