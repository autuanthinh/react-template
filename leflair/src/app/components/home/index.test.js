import React from 'react';
import TestRenderer from 'react-test-renderer';

import Home from './index';

describe('Home component', () => {
    it('render correctly', () => {
        const tree = TestRenderer.create(<Home />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
