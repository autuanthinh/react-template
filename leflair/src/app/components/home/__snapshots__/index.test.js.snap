// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component render correctly 1`] = `
<div
  id="homePage"
>
  <div
    className="container"
  >
    <h4
      className="title"
    >
      Guide
    </h4>
    <img
      alt="architect"
      src="architect.png"
    />
    <div
      className="mainContent"
    >
      <div
        className="eachRow"
      >
        <strong>
          Operating system: 
        </strong>
        <span>
          Mac OS
        </span>
      </div>
      <div
        className="eachRow"
      >
        <dl>
          <dt>
            <strong>
              Install: 
            </strong>
          </dt>
          <dd>
            - Open Terminal, cd leflair project, type 
            <strong>
              yarn install
            </strong>
             or 
            <strong>
              npm i
            </strong>
          </dd>
          <dd>
            - start project: type 
            <strong>
              yarn start-dev
            </strong>
             or 
            <strong>
              npm run start-dev
            </strong>
          </dd>
          <dd>
            - build project: type 
            <strong>
              yarn build
            </strong>
             or 
            <strong>
              npm run build
            </strong>
          </dd>
        </dl>
      </div>
      <div
        className="eachRow"
      >
        <strong>
          Design: 
        </strong>
        <span>
          Support mobile, table, pc
        </span>
      </div>
      <div
        className="eachRow"
      >
        <dl>
          <dt>
            <strong>
              <strong>
                Structure : 
              </strong>
               
            </strong>
          </dt>
          <dd>
            - As image above, all techniques that i used, and some techniques i add more as redux-form, react-lazyload, etc...
          </dd>
          <dd>
            - your cart menu, when quantity more than 0 must be able to click
          </dd>
          <dd>
            - chrome tab network (js), check slipt file bundle.js when user click menu on top right (avoid all file js build into one bundle.js file, when scale up project, this is very useful)
          </dd>
        </dl>
      </div>
    </div>
  </div>
</div>
`;
